pysteps/blending/steps.py

```diff
@@ -65,1723 +64,2858 @@
+            # Rainfall outside the pure extrapolation domain is not taken into account.
+            if np.any(np.isfinite(worker_state.final_blended_forecast_recomposed)):
+                worker_state.final_blended_forecast_recomposed = (
+                    probmatching.nonparam_match_empirical_cdf(
+                        worker_state.final_blended_forecast_recomposed,
+                        precip_forecast_probability_matching_resampled,
+                        nan_indices,
+                    )
+                )
+                precip_forecast_probability_matching_resampled = None
+        elif self.__config.probmatching_method == "mean":
+            # Use R_pm_blended as benchmark field and
+            mean_probabiltity_matching_forecast = np.mean(
+                precip_forecast_probability_matching_resampled[
+                    precip_forecast_probability_matching_resampled
+                    >= self.__config.precip_threshold
+                ]
+            )
+            no_rain_mask = (
+                worker_state.final_blended_forecast_recomposed
+                >= self.__config.precip_threshold
+            )
+            mean_precip_forecast = np.mean(
+                worker_state.final_blended_forecast_recomposed[no_rain_mask]
+            )
+            worker_state.final_blended_forecast_recomposed[no_rain_mask] = (
+                worker_state.final_blended_forecast_recomposed[no_rain_mask]
+                - mean_precip_forecast
+                + mean_probabiltity_matching_forecast
+            )
+            precip_forecast_probability_matching_resampled = None
+
+        final_blended_forecast_single_member.append(
+            worker_state.final_blended_forecast_recomposed
+        )
+        return final_blended_forecast_single_member
+
+    def __measure_time(self, label, start_time):
+        """
+        Measure and print the time taken for a specific part of the process.
+
+        Parameters:
+        - label: A description of the part of the process being measured.
+        - start_time: The timestamp when the process started (from time.time()).
+        """
+        if self.__config.measure_time:
+            elapsed_time = time.time() - start_time
+            print(f"{label} took {elapsed_time:.2f} seconds.")
+            return elapsed_time
+        return None
+
+    def reset_states_and_params(self):
+        """
+        Reset the internal state and parameters of the nowcaster to allow multiple forecasts.
+        This method resets the state and params to their initial conditions without reinitializing
+        the inputs like precip, velocity, time_steps, or config.
+        """
+        # Re-initialize the state and parameters
+        self.__state = StepsBlendingState()
+        self.__params = StepsBlendingParams()
+
+        # Reset time measurement variables
+        self.__start_time_init = None
+        self.__init_time = None
+        self.__mainloop_time = None
```

Comment:
Does this not only make sense if you specify `precip` and `velocity` (and `precip_models` and `velocity_models`) when you call `compute_forecast`, because otherwise you would only be able to rerun the exact same forecast with the same inputs again?

Reply:
I followed the same logic here as for the nowcaster but maybe that is not optimal for the nowcaster either. The idea is if people want to generate a bunch of forecasts, they can make one config and then loop over all forecast they want to do, each forecast makes a new blendednowcaster which then computed the forecasts. But indeed, in that scenario, we don't need the reset_states... What do you think would be the best way forward:
- Remove this reset and keep the rest as is
- Let people create a forecaster once and add the precip and velocity fields as parameters to the compute_forecast()
- Something else

Reply:
Stylistically I like the idea of a reusable nowcast instance that you initialize once with a config and then can call multiple times with different precip and velocity fields (so option 2).

Practically though option 1 might be more robust. If you create a new nowcast instance every time that means that the old one (and all data in it) can be cleaned up by the garbage collector. If you keep the nowcast instance around for an extended time (in a production setting for example) you rely on the reset state method to clear everything, so that needs to be called every time after a nowcast has completed and the reset state method **has** to reset the state properly every time. So data running over from one run into the other or memory leaks happening are more likely with option 2.

Also the way that it is done now, where everything (including the inputs) are just stored in `self` and nothing is passed between methods using arguments fits better with option 1 I think.

Reply:
Okay, I will remove the reset funtion and add some comments as to how the user is expected to deal with the class.